pallets/messages/src/lib.rs

```diff
@@ -117,29 +124,27 @@ pub mod pallet {
 	}
 
 	#[pallet::pallet]
+	#[pallet::storage_version(STORAGE_VERSION)]
 	pub struct Pallet<T>(_);
 
-	/// A permanent storage for messages mapped by block number and schema id.
-	/// - Keys: BlockNumber, Schema Id
-	/// - Value: List of Messages
 	#[pallet::storage]
-	#[pallet::getter(fn get_messages)]
-	pub(super) type Messages<T: Config> = StorageDoubleMap<
-		_,
-		Twox64Concat,
-		BlockNumberFor<T>,
-		Twox64Concat,
-		SchemaId,
-		BoundedVec<Message<T::MessagesMaxPayloadSizeBytes>, T::MaxMessagesPerBlock>,
-		ValueQuery,
-	>;
+	#[pallet::whitelist_storage]
+	#[pallet::getter(fn get_block_metadata)]
+	pub(super) type MessageBlockMetadata<T: Config> =
+		StorageValue<_, BlockMetadata<T::MaxMessagesPerBlock>, ValueQuery>;
```

Comment:
I think we should consider using 
https://crates.parity.io/frame_support/storage/bounded_btree_set/struct.BoundedBTreeSet.html

I think it will simplify the code a little.

Reply:
Not applicable anymore